Enable regression tests of weak memory model instrumentation

tautschnig opened the pull request:
These regression tests are a selected subset (cf. README.txt) of the
full set of available tests (stored in goto-instrument-wmm-full.tgz),
and will ensure basic testing of the functionality implemented in
goto-instrument.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Enable regression tests of weak memory model instrumentation

These regression tests are a selected subset (cf. README.txt) of the
full set of available tests (stored in goto-instrument-wmm-full.tgz),
and will ensure basic testing of the functionality implemented in
goto-instrument. Test specifications were updated to work with current
infrastructure.

tautschnig committed:
Add regression tests from goto-instrument-wmm-full.tgz

Make all regression tests from the archive available (yet marking them as
THOROUGH or KNOWNBUG to avoid including them by default). This avoids having an
unused archive file bit-rotting in the repository. This commit was generated as
follows:

```

set -e

cd regression/goto-instrument-wmm-core/
tar xzf ../goto-instrument-wmm-full.tgz --strip-components=1
for f in $(tar tzf ../goto-instrument-wmm-full.tgz | sed 's#^goto-instrument-wmm-full/##')
do
  [ -f $f ] || continue
  if git status --porcelain $f | grep -q '^?'
  then
    if [[ $(basename $f) == "test.desc" ]]
    then
      sed -i '1s/^CORE$/THOROUGH/' $f
      if ! grep -q EXIT $f
      then
        if grep -q "VERIFICATION FAILED" $f
        then
          sed -i '/SIGNAL=0/ i ^EXIT=10$' $f
        else
          sed -i '/SIGNAL=0/ i ^EXIT=0$' $f
        fi
      fi
    elif [[ $f =~ \.c$ ]]
    then
      clang-format -i $f
    fi
    git add $f
  else
    git checkout -- $f
  fi
done
git rm ../goto-instrument-wmm-full.tgz

git commit -F - <<EOF
Add regression tests from goto-instrument-wmm-full.tgz

Make all regression tests from the archive available (yet marking them as
THOROUGH or KNOWNBUG to avoid including them by default). This avoids having an
unused archive file bit-rotting in the repository. This commit was generated as
follows:

\`\`\`
$(<../../script.sh)
\`\`\`
EOF
```

tautschnig committed:
goto-instrument-wmm tests: minimisation requires glpk

Tag all *_OPT tests "glpk" and exclude them from test runs by default as
we don't currently build with glpk by default.

tautschnig committed:
goto-instrument-wmm tests: mark failing tests KNOWNBUG

Some of the tests that were previously imported as THOROUGH from the tar
archive actually are failing and require further investigation.

tautschnig committed:
goto-instrument-wmm tests: flip "THOROUGH" to "CORE"

These tests currently pass and each of them completes in under 1 second.

martin-cs reviewed (approved):
I must confess that I haven't read every line of this PR but I think I get the general picture.  One concern : how often do the THOROUGH tests get run.  Could we add it to the new version / release workflow?

tautschnig committed:
goto-instrument-wmm: removing graph edges invalidates the iterator

Store the elements to be removed in a set until after the iteration is
done, and then remove the chosen elements. The more standard approach
could be returning the iterator to the next element in the
remove-an-element function, but this requires larger changes here. Thus
keeping it small and simple for now.

tautschnig commented:
> I must confess that I haven't read every line of this PR but I think I get the general picture. One concern : how often do the THOROUGH tests get run. Could we add it to the new version / release workflow?

I'll create a separate PR for this.
